### PR TITLE
[2.7] bpo-36742: Fixes handling of pre-normalization characters in urlsplit() (GH-13017)

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue36742

### CODE REVIEW

File: Lib/test/test_urlparse.py [hunk 1]
```diff
@@ -641,6 +641,12 @@ def test_urlsplit_normalization(self):
         self.assertIn(u'\u2100', denorm_chars)
         self.assertIn(u'\uFF03', denorm_chars)
 
+        # bpo-36742: Verify port separators are ignored when they
+        # existed prior to decomposition
+        urllib.parse.urlsplit('http://\u30d5\u309a:80')
```
Comment (tirkarthi) [inline Lib/test/test_urlparse.py]: I guess the Python 3 patch applied here. `urlunparse.urlsplit` could be the correct function for this test.

Comment (zooba) [reply Lib/test/test_urlparse.py]: Yep, you're right. Thanks!

Comment (tirkarthi) [reply Lib/test/test_urlparse.py]: You're welcome. There is some test error on CI :

```
test test_urlparse failed -- Traceback (most recent call last):
  File "/home/travis/build/python/cpython/Lib/test/test_urlparse.py", line 648, in test_urlsplit_normalization
    urlparse.urlsplit('http://\u30d5\u309a\ufe1380')
AssertionError: ValueError not raised
```

Comment (tirkarthi) [reply Lib/test/test_urlparse.py]: Ah I guess u"" prefix is missing since it's Python 2 ? Adding it makes the tests to pass locally for the PR.

**Review by tirkarthi — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**